src/main/java/com/amazon/opendistroforelasticsearch/security/auditlog/AuditConfig.java

```diff
@@ -0,0 +1,441 @@
+
+        List<String> ignoredComplianceUsersForWrite = getConfigList(
+                settings,
+                ConfigConstants.OPENDISTRO_SECURITY_COMPLIANCE_HISTORY_WRITE_IGNORE_USERS,
+                defaultIgnoredUsers);
+
+        List<String> ignoreAuditRequests = getConfigList(
+                settings,
+                ConfigConstants.OPENDISTRO_SECURITY_AUDIT_IGNORE_REQUESTS,
+                Collections.emptyList());
+
+        List<String> watchedReadFields = settings.getAsList(
+                ConfigConstants.OPENDISTRO_SECURITY_COMPLIANCE_HISTORY_READ_WATCHED_FIELDS,
+                Collections.emptyList(),
+                false);
+
+        List<String> watchedWriteIndices = settings.getAsList(
+                ConfigConstants.OPENDISTRO_SECURITY_COMPLIANCE_HISTORY_WRITE_WATCHED_INDICES,
+                Collections.emptyList());
+
+        Set<String> immutableIndicesPatterns = new HashSet<>(settings.getAsList(ConfigConstants.OPENDISTRO_SECURITY_COMPLIANCE_IMMUTABLE_INDICES, Collections.emptyList()));
+
+        String saltAsString = settings.get(ConfigConstants.OPENDISTRO_SECURITY_COMPLIANCE_SALT, ConfigConstants.OPENDISTRO_SECURITY_COMPLIANCE_SALT_DEFAULT);
+
+        String opendistrosecurityIndex = settings.get(ConfigConstants.OPENDISTRO_SECURITY_CONFIG_INDEX_NAME, ConfigConstants.OPENDISTRO_SECURITY_DEFAULT_CONFIG_INDEX);
+
+        String type = settings.get(ConfigConstants.OPENDISTRO_SECURITY_AUDIT_TYPE_DEFAULT, null);
+
+        String index = settings.get(ConfigConstants.OPENDISTRO_SECURITY_AUDIT_CONFIG_DEFAULT_PREFIX + ConfigConstants.OPENDISTRO_SECURITY_AUDIT_ES_INDEX, "'security-auditlog-'YYYY.MM.dd");
+
+        return new AuditConfig(restAuditingEnabled,
+                transportAuditingEnabled,
+                resolveBulkRequests,
+                logRequestBody,
+                resolveIndices,
+                excludeSensitiveHeaders,
+                logDiffsForWrite,
+                logWriteMetadataOnly,
+                logReadMetadataOnly,
+                logExternalConfig,
+                logInternalConfig,
+                ignoredAuditUsers,
+                ignoredComplianceUsersForRead,
+                ignoredComplianceUsersForWrite,
+                ignoreAuditRequests,
+                watchedReadFields,
+                watchedWriteIndices,
+                disabledRestCategories,
+                disabledTransportCategories,
+                immutableIndicesPatterns,
+                saltAsString,
+                opendistrosecurityIndex,
+                type,
+                index);
+    }
+
+    private static List<String> getConfigList(final Settings settings,
+                                              final String key,
+                                              final List<String> defaultList) {
+        List<String> list = new ArrayList<>(settings.getAsList(key, defaultList));
+        if (list.size() == 1 && "NONE".equals(list.get(0))) {
```

Comment:
Following up from previous conversation, users can do any of the following
1) list: []
2) list: ["NONE"]
3) list: "NONE"

Is there a reason why NONE must not be supported as a valid option ? 

I m worried removing this will cause a breaking change for users during upgrade paths (who ever is using NONE currently) and will have to explicitly document this even for backports.

Reply:
It seems redundant to have three different ways to specify the same thing. I understand the need for backward compatibility. It can be your decision if you want to support this in the future as well.

Reply:
Well I m not supporting or introducing this feature here.
Rather I m not breaking the current support